libarchive/test/test_short_writes.c

```diff
@@ -49,14 +49,14 @@ struct checker {
 static ssize_t
 short_write_callback(struct archive *a, void *client_data, const void *buffer, size_t length)
 {
-        (void)a;
-
         struct checker *checker = client_data;
-        size_t to_write = length < 100 ? length : 100;
-        size_t new_len = checker->shortbuf_len + to_write;
+        const ssize_t to_write = length < 100 ? (ssize_t)length : 100;
```

Comment:
Unrelated `size_t` -> `ssize_t` - please drop and keep a separate patch, with clear commit message explaining why.